File: demos/routing/lib/libsodium-port/bench/benchMult.js
var BENCHCOUNT = 100000;

// require c++ binding module
var wrapper = require('../wrapper.js');
var BN = require('bn.js');

wrapper.ready.then(function () {
  // Find and hash some Point
  var point = wrapper.hashToPoint('hello');

  // Find scalar
  var random = new BN(Math.floor(Math.random() * 10000));
  var scalar = wrapper.BNToBytes(random);

  // Run several tests
  for (var t = 0; t < BENCHCOUNT; t++) {
    point = wrapper.scalarMult(point, scalar);
  }

  console.log('success mult');
});

File: demos/routing/lib/libsodium-port/bench/benchMult.c
#define BENCHCOUNT 100000

#include <sodium.h>
#include <stdio.h>

void main() {
  if (sodium_init() == -1) {
    return;
  }

  unsigned char hash[crypto_generichash_BYTES];
  unsigned char point[crypto_scalarmult_ed25519_BYTES];
  unsigned char result[crypto_scalarmult_ed25519_BYTES];
  unsigned char scalar[crypto_scalarmult_ed25519_SCALARBYTES];

  // Hash
  crypto_generichash(hash, sizeof hash, "hello", 5, NULL, 0);
  crypto_core_ed25519_from_uniform(point, hash);

  // Benchmarks
  for(int i = 0; i < BENCHCOUNT; i++) {
    randombytes_buf(scalar, sizeof scalar);
    if (crypto_scalarmult_ed25519_noclamp(result, scalar, point) != 0) {
      printf("FAIL SCALAR MULT\n");
      return;
    }
  }

  printf("success scalarMult\n");
}


